core/src/scene/sceneLoader.cpp

```diff
@@ -733,58 +744,67 @@ Filter SceneLoader::generateNoneFilter(YAML::Node _filter) {
+std::vector<StyleParam> SceneLoader::parseStyleParams(Node params, const std::string& prefix) {
 
-    for (const auto& propItr : styleProps) {
-        Node prop = propItr.first;
-        std::string paramKey;
-        if (propPrefix.length() > 0) {
-            paramKey = propPrefix + ":" + prop.as<std::string>();
-        } else {
-            paramKey = prop.as<std::string>();
-        }
-        if (propItr.second.IsScalar()) {
-            paramMap.emplace(paramKey, propItr.second.as<std::string>());
-        } else if (propItr.second.IsSequence()) {
-            paramMap.emplace(paramKey, parseSequence(propItr.second));
-        } else if (propItr.second.IsMap()) {
-            parseStyleProps(propItr.second, paramMap, paramKey);
-        } else {
-            logMsg("Error: Badly formed Style property, need to be a scalar, sequence or map."
-                    "%s will not be added to stype properties.\n", paramKey.c_str());
+    std::vector<StyleParam> out;
+
+    for (const auto& prop : params) {
+
+        std::string key = (prefix.empty() ? "" : (prefix + ":")) + prop.first.as<std::string>();
+
+        Node value = prop.second;
+
+        switch (value.Type()) {
+            case YAML::NodeType::Scalar:   out.push_back({ key, value.as<std::string>() }); break;
+            case YAML::NodeType::Sequence: out.push_back({ key, parseSequence(value) }); break;
+            case YAML::NodeType::Map: {
+                auto subparams = parseStyleParams(value, key);
+                out.insert(out.end(), subparams.begin(), subparams.end());
+                break;
+            }
+            default: logMsg("ERROR: Style parameter %s must be a scalar, sequence, or map.\n", key.c_str());
         }
     }
 
-}
-
-void SceneLoader::loadSublayers(YAML::Node layer, std::vector<std::shared_ptr<SceneLayer>>& subLayers) {
+    return out;
 
-    for (const auto& subLayerItr : layer) {
+}
 
-        std::string subLayerName = subLayerItr.first.as<std::string>();
+SceneLayer SceneLoader::loadSublayer(YAML::Node layer, const std::string& name, Scene& scene) {
 
-        Node drawGroup = subLayerItr.second["draw"];
+    std::vector<SceneLayer> sublayers;
+    std::vector<DrawRule> rules;
+    Filter filter;
 
-        // If node has a draw group that means its a sublayer (no need to check for reserved)
-        if(drawGroup) {
+    for (const auto& member : layer) {
 
-            std::vector<std::shared_ptr<SceneLayer>> ssubLayers;
-            Node filter = subLayerItr.second["filter"];
-            Filter subLayerFilter = generateFilter(filter);
+        const std::string key = member.first.as<std::string>();
 
-            loadSublayers(layer[subLayerName], ssubLayers);
+        if (key == "data") {
+            // Ignored for sublayers
+        } else if (key == "draw") {
+            // Member is a mapping of draw rules
+            for (auto& ruleNode : member.second) {
```

Comment:
Correct me if I am wrong here.
According to what I get from the drawGroup [documentation](https://github.com/tangrams/tangram-docs/blob/gh-pages/pages/draw.md#draw), there can only be one drawGroup per layer/subLayer and rules defined in one layer are passed down sublayers.

Can you give me an example of how size of "rules" here can ever be more than one?